Description: avrdude fuse read mask change
Author: Milan Kupcevic <milan@debian.org>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=824815
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/hardware/arduino/boards.txt
+++ b/hardware/arduino/boards.txt
@@ -8,11 +8,11 @@
 uno.upload.speed=115200
 uno.bootloader.low_fuses=0xff
 uno.bootloader.high_fuses=0xde
-uno.bootloader.extended_fuses=0x05
+uno.bootloader.extended_fuses=0xfd
 uno.bootloader.path=optiboot
 uno.bootloader.file=optiboot_atmega328.hex
-uno.bootloader.unlock_bits=0x3F
-uno.bootloader.lock_bits=0x0F
+uno.bootloader.unlock_bits=0xff
+uno.bootloader.lock_bits=0xcf
 uno.build.mcu=atmega328p
 uno.build.f_cpu=16000000L
 uno.build.core=arduino
@@ -28,11 +28,11 @@
 
 atmega328.bootloader.low_fuses=0xFF
 atmega328.bootloader.high_fuses=0xDA
-atmega328.bootloader.extended_fuses=0x05
+atmega328.bootloader.extended_fuses=0xfd
 atmega328.bootloader.path=atmega
 atmega328.bootloader.file=ATmegaBOOT_168_atmega328.hex
-atmega328.bootloader.unlock_bits=0x3F
-atmega328.bootloader.lock_bits=0x0F
+atmega328.bootloader.unlock_bits=0xff
+atmega328.bootloader.lock_bits=0xcf
 
 atmega328.build.mcu=atmega328p
 atmega328.build.f_cpu=16000000L
@@ -49,11 +49,11 @@
 
 diecimila.bootloader.low_fuses=0xff
 diecimila.bootloader.high_fuses=0xdd
-diecimila.bootloader.extended_fuses=0x00
+diecimila.bootloader.extended_fuses=0xf8
 diecimila.bootloader.path=atmega
 diecimila.bootloader.file=ATmegaBOOT_168_diecimila.hex
-diecimila.bootloader.unlock_bits=0x3F
-diecimila.bootloader.lock_bits=0x0F
+diecimila.bootloader.unlock_bits=0xff
+diecimila.bootloader.lock_bits=0xcf
 
 diecimila.build.mcu=atmega168
 diecimila.build.f_cpu=16000000L
@@ -70,11 +70,11 @@
 
 nano328.bootloader.low_fuses=0xFF
 nano328.bootloader.high_fuses=0xDA
-nano328.bootloader.extended_fuses=0x05
+nano328.bootloader.extended_fuses=0xfd
 nano328.bootloader.path=atmega
 nano328.bootloader.file=ATmegaBOOT_168_atmega328.hex
-nano328.bootloader.unlock_bits=0x3F
-nano328.bootloader.lock_bits=0x0F
+nano328.bootloader.unlock_bits=0xff
+nano328.bootloader.lock_bits=0xcf
 
 nano328.build.mcu=atmega328p
 nano328.build.f_cpu=16000000L
@@ -91,11 +91,11 @@
 
 nano.bootloader.low_fuses=0xff
 nano.bootloader.high_fuses=0xdd
-nano.bootloader.extended_fuses=0x00
+nano.bootloader.extended_fuses=0xf8
 nano.bootloader.path=atmega
 nano.bootloader.file=ATmegaBOOT_168_diecimila.hex
-nano.bootloader.unlock_bits=0x3F
-nano.bootloader.lock_bits=0x0F
+nano.bootloader.unlock_bits=0xff
+nano.bootloader.lock_bits=0xcf
 
 nano.build.mcu=atmega168
 nano.build.f_cpu=16000000L
@@ -115,8 +115,8 @@
 mega2560.bootloader.extended_fuses=0xFD
 mega2560.bootloader.path=stk500v2
 mega2560.bootloader.file=stk500boot_v2_mega2560.hex
-mega2560.bootloader.unlock_bits=0x3F
-mega2560.bootloader.lock_bits=0x0F
+mega2560.bootloader.unlock_bits=0xff
+mega2560.bootloader.lock_bits=0xcf
 
 mega2560.build.mcu=atmega2560
 mega2560.build.f_cpu=16000000L
@@ -136,8 +136,8 @@
 mega.bootloader.extended_fuses=0xF5
 mega.bootloader.path=atmega
 mega.bootloader.file=ATmegaBOOT_168_atmega1280.hex
-mega.bootloader.unlock_bits=0x3F
-mega.bootloader.lock_bits=0x0F
+mega.bootloader.unlock_bits=0xff
+mega.bootloader.lock_bits=0xcf
 
 mega.build.mcu=atmega1280
 mega.build.f_cpu=16000000L
@@ -156,8 +156,8 @@
 leonardo.bootloader.extended_fuses=0xcb
 leonardo.bootloader.path=caterina
 leonardo.bootloader.file=Caterina-Leonardo.hex
-leonardo.bootloader.unlock_bits=0x3F
-leonardo.bootloader.lock_bits=0x2F
+leonardo.bootloader.unlock_bits=0xff
+leonardo.bootloader.lock_bits=0xef
 leonardo.build.mcu=atmega32u4
 leonardo.build.f_cpu=16000000L
 leonardo.build.vid=0x2341
@@ -177,8 +177,8 @@
 esplora.bootloader.extended_fuses=0xcb
 esplora.bootloader.path=caterina
 esplora.bootloader.file=Caterina-Esplora.hex
-esplora.bootloader.unlock_bits=0x3F
-esplora.bootloader.lock_bits=0x2F
+esplora.bootloader.unlock_bits=0xff
+esplora.bootloader.lock_bits=0xef
 esplora.build.mcu=atmega32u4
 esplora.build.f_cpu=16000000L
 esplora.build.vid=0x2341
@@ -198,8 +198,8 @@
 micro.bootloader.extended_fuses=0xcb
 micro.bootloader.path=caterina
 micro.bootloader.file=Caterina-Micro.hex
-micro.bootloader.unlock_bits=0x3F
-micro.bootloader.lock_bits=0x2F
+micro.bootloader.unlock_bits=0xff
+micro.bootloader.lock_bits=0xef
 micro.build.mcu=atmega32u4
 micro.build.f_cpu=16000000L
 micro.build.vid=0x2341
@@ -217,11 +217,11 @@
 
 mini328.bootloader.low_fuses=0xff
 mini328.bootloader.high_fuses=0xd8
-mini328.bootloader.extended_fuses=0x05
+mini328.bootloader.extended_fuses=0xfd
 mini328.bootloader.path=optiboot
 mini328.bootloader.file=optiboot_atmega328-Mini.hex
-mini328.bootloader.unlock_bits=0x3F
-mini328.bootloader.lock_bits=0x0F
+mini328.bootloader.unlock_bits=0xff
+mini328.bootloader.lock_bits=0xcf
 
 mini328.build.mcu=atmega328p
 mini328.build.f_cpu=16000000L
@@ -238,11 +238,11 @@
 
 mini.bootloader.low_fuses=0xff
 mini.bootloader.high_fuses=0xdd
-mini.bootloader.extended_fuses=0x00
+mini.bootloader.extended_fuses=0xf8
 mini.bootloader.path=atmega
 mini.bootloader.file=ATmegaBOOT_168_ng.hex
-mini.bootloader.unlock_bits=0x3F
-mini.bootloader.lock_bits=0x0F
+mini.bootloader.unlock_bits=0xff
+mini.bootloader.lock_bits=0xcf
 
 mini.build.mcu=atmega168
 mini.build.f_cpu=16000000L
@@ -259,11 +259,11 @@
 
 ethernet.bootloader.low_fuses=0xff
 ethernet.bootloader.high_fuses=0xde
-ethernet.bootloader.extended_fuses=0x05
+ethernet.bootloader.extended_fuses=0xfd
 ethernet.bootloader.path=optiboot
 ethernet.bootloader.file=optiboot_atmega328.hex
-ethernet.bootloader.unlock_bits=0x3F
-ethernet.bootloader.lock_bits=0x0F
+ethernet.bootloader.unlock_bits=0xff
+ethernet.bootloader.lock_bits=0xcf
 
 ethernet.build.variant=standard
 ethernet.build.mcu=atmega328p
@@ -280,11 +280,11 @@
 
 fio.bootloader.low_fuses=0xFF
 fio.bootloader.high_fuses=0xDA
-fio.bootloader.extended_fuses=0x05
+fio.bootloader.extended_fuses=0xfd
 fio.bootloader.path=arduino:atmega
 fio.bootloader.file=ATmegaBOOT_168_atmega328_pro_8MHz.hex
-fio.bootloader.unlock_bits=0x3F
-fio.bootloader.lock_bits=0x0F
+fio.bootloader.unlock_bits=0xff
+fio.bootloader.lock_bits=0xcf
 
 fio.build.mcu=atmega328p
 fio.build.f_cpu=8000000L
@@ -302,11 +302,11 @@
 
 bt328.bootloader.low_fuses=0xff
 bt328.bootloader.high_fuses=0xd8
-bt328.bootloader.extended_fuses=0x05
+bt328.bootloader.extended_fuses=0xfd
 bt328.bootloader.path=bt
 bt328.bootloader.file=ATmegaBOOT_168_atmega328_bt.hex
-bt328.bootloader.unlock_bits=0x3F
-bt328.bootloader.lock_bits=0x0F
+bt328.bootloader.unlock_bits=0xff
+bt328.bootloader.lock_bits=0xcf
 
 bt328.build.mcu=atmega328p
 bt328.build.f_cpu=16000000L
@@ -324,11 +324,11 @@
 
 bt.bootloader.low_fuses=0xff
 bt.bootloader.high_fuses=0xdd
-bt.bootloader.extended_fuses=0x00
+bt.bootloader.extended_fuses=0xf8
 bt.bootloader.path=bt
 bt.bootloader.file=ATmegaBOOT_168.hex
-bt.bootloader.unlock_bits=0x3F
-bt.bootloader.lock_bits=0x0F
+bt.bootloader.unlock_bits=0xff
+bt.bootloader.lock_bits=0xcf
 
 bt.build.mcu=atmega168
 bt.build.f_cpu=16000000L
@@ -347,8 +347,8 @@
 LilyPadUSB.bootloader.extended_fuses=0xce
 LilyPadUSB.bootloader.path=caterina-LilyPadUSB
 LilyPadUSB.bootloader.file=Caterina-LilyPadUSB.hex
-LilyPadUSB.bootloader.unlock_bits=0x3F
-LilyPadUSB.bootloader.lock_bits=0x2F
+LilyPadUSB.bootloader.unlock_bits=0xff
+LilyPadUSB.bootloader.lock_bits=0xef
 LilyPadUSB.build.mcu=atmega32u4
 LilyPadUSB.build.f_cpu=8000000L
 LilyPadUSB.build.vid=0x1B4F
@@ -366,11 +366,11 @@
 
 lilypad328.bootloader.low_fuses=0xFF
 lilypad328.bootloader.high_fuses=0xDA
-lilypad328.bootloader.extended_fuses=0x05
+lilypad328.bootloader.extended_fuses=0xfd
 lilypad328.bootloader.path=atmega
 lilypad328.bootloader.file=ATmegaBOOT_168_atmega328_pro_8MHz.hex
-lilypad328.bootloader.unlock_bits=0x3F
-lilypad328.bootloader.lock_bits=0x0F
+lilypad328.bootloader.unlock_bits=0xff
+lilypad328.bootloader.lock_bits=0xcf
 
 lilypad328.build.mcu=atmega328p
 lilypad328.build.f_cpu=8000000L
@@ -387,11 +387,11 @@
 
 lilypad.bootloader.low_fuses=0xe2
 lilypad.bootloader.high_fuses=0xdd
-lilypad.bootloader.extended_fuses=0x00
+lilypad.bootloader.extended_fuses=0xf8
 lilypad.bootloader.path=lilypad
 lilypad.bootloader.file=LilyPadBOOT_168.hex
-lilypad.bootloader.unlock_bits=0x3F
-lilypad.bootloader.lock_bits=0x0F
+lilypad.bootloader.unlock_bits=0xff
+lilypad.bootloader.lock_bits=0xcf
 
 lilypad.build.mcu=atmega168
 lilypad.build.f_cpu=8000000L
@@ -408,11 +408,11 @@
 
 pro5v328.bootloader.low_fuses=0xFF
 pro5v328.bootloader.high_fuses=0xDA
-pro5v328.bootloader.extended_fuses=0x05
+pro5v328.bootloader.extended_fuses=0xfd
 pro5v328.bootloader.path=atmega
 pro5v328.bootloader.file=ATmegaBOOT_168_atmega328.hex
-pro5v328.bootloader.unlock_bits=0x3F
-pro5v328.bootloader.lock_bits=0x0F
+pro5v328.bootloader.unlock_bits=0xff
+pro5v328.bootloader.lock_bits=0xcf
 
 pro5v328.build.mcu=atmega328p
 pro5v328.build.f_cpu=16000000L
@@ -429,11 +429,11 @@
 
 pro5v.bootloader.low_fuses=0xff
 pro5v.bootloader.high_fuses=0xdd
-pro5v.bootloader.extended_fuses=0x00
+pro5v.bootloader.extended_fuses=0xf8
 pro5v.bootloader.path=atmega
 pro5v.bootloader.file=ATmegaBOOT_168_diecimila.hex
-pro5v.bootloader.unlock_bits=0x3F
-pro5v.bootloader.lock_bits=0x0F
+pro5v.bootloader.unlock_bits=0xff
+pro5v.bootloader.lock_bits=0xcf
 
 pro5v.build.mcu=atmega168
 pro5v.build.f_cpu=16000000L
@@ -450,11 +450,11 @@
 
 pro328.bootloader.low_fuses=0xFF
 pro328.bootloader.high_fuses=0xDA
-pro328.bootloader.extended_fuses=0x05
+pro328.bootloader.extended_fuses=0xfd
 pro328.bootloader.path=atmega
 pro328.bootloader.file=ATmegaBOOT_168_atmega328_pro_8MHz.hex
-pro328.bootloader.unlock_bits=0x3F
-pro328.bootloader.lock_bits=0x0F
+pro328.bootloader.unlock_bits=0xff
+pro328.bootloader.lock_bits=0xcf
 
 pro328.build.mcu=atmega328p
 pro328.build.f_cpu=8000000L
@@ -471,11 +471,11 @@
 
 pro.bootloader.low_fuses=0xc6
 pro.bootloader.high_fuses=0xdd
-pro.bootloader.extended_fuses=0x00
+pro.bootloader.extended_fuses=0xf8
 pro.bootloader.path=atmega
 pro.bootloader.file=ATmegaBOOT_168_pro_8MHz.hex
-pro.bootloader.unlock_bits=0x3F
-pro.bootloader.lock_bits=0x0F
+pro.bootloader.unlock_bits=0xff
+pro.bootloader.lock_bits=0xcf
 
 pro.build.mcu=atmega168
 pro.build.f_cpu=8000000L
@@ -492,11 +492,11 @@
 
 atmega168.bootloader.low_fuses=0xff
 atmega168.bootloader.high_fuses=0xdd
-atmega168.bootloader.extended_fuses=0x00
+atmega168.bootloader.extended_fuses=0xf8
 atmega168.bootloader.path=atmega
 atmega168.bootloader.file=ATmegaBOOT_168_ng.hex
-atmega168.bootloader.unlock_bits=0x3F
-atmega168.bootloader.lock_bits=0x0F
+atmega168.bootloader.unlock_bits=0xff
+atmega168.bootloader.lock_bits=0xcf
 
 atmega168.build.mcu=atmega168
 atmega168.build.f_cpu=16000000L
@@ -515,8 +515,8 @@
 atmega8.bootloader.high_fuses=0xca
 atmega8.bootloader.path=atmega8
 atmega8.bootloader.file=ATmegaBOOT-prod-firmware-2009-11-07.hex
-atmega8.bootloader.unlock_bits=0x3F
-atmega8.bootloader.lock_bits=0x0F
+atmega8.bootloader.unlock_bits=0xff
+atmega8.bootloader.lock_bits=0xcf
 
 atmega8.build.mcu=atmega8
 atmega8.build.f_cpu=16000000L
@@ -535,8 +535,8 @@
 robotControl.bootloader.extended_fuses=0xcb
 robotControl.bootloader.path=caterina-Arduino_Robot
 robotControl.bootloader.file=Caterina-Robot-Control.hex
-robotControl.bootloader.unlock_bits=0x3F
-robotControl.bootloader.lock_bits=0x2F
+robotControl.bootloader.unlock_bits=0xff
+robotControl.bootloader.lock_bits=0xef
 robotControl.build.mcu=atmega32u4
 robotControl.build.f_cpu=16000000L
 robotControl.build.vid=0x2341
@@ -556,8 +556,8 @@
 robotMotor.bootloader.extended_fuses=0xcb
 robotMotor.bootloader.path=caterina-Arduino_Robot
 robotMotor.bootloader.file=Caterina-Robot-Motor.hex
-robotMotor.bootloader.unlock_bits=0x3F
-robotMotor.bootloader.lock_bits=0x2F
+robotMotor.bootloader.unlock_bits=0xff
+robotMotor.bootloader.lock_bits=0xef
 robotMotor.build.mcu=atmega32u4
 robotMotor.build.f_cpu=16000000L
 robotMotor.build.vid=0x2341
